### PR TITLE
CRUD prototype #2

### PR DESCRIPTION
**Pull request by ErikGmz**
Created basic event CRUD, defined all EventsController methods and paths.

Captura de pantalla 2024-01-24 193202
Captura de pantalla 2024-01-24 193237
image
image

### CODE REVIEW

File: app/views/events/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<h1>Edit event</h1>
+<%= render 'form', event: @event %>
+<br />
```
Comment (juperameza) [inline app/views/events/edit.html.erb]: DOnt use br 
Is a bad practice

File: app/views/events/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<h1>New event</h1>
+<%= render 'form', event: @event %>
+<br />
```
Comment (juperameza) [inline app/views/events/new.html.erb]: dont ue br

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h2>Event details</h2>
+<%= render 'event', event: @event %>
+<br />
```
Comment (juperameza) [inline app/views/events/show.html.erb]: dont use br

File: config/routes.rb [hunk 4]
```diff
@@ -6,5 +8,14 @@
...
+  root 'events#index'
+
+  get '/events', to: 'events#index', as: 'events'
+  post '/events', to: 'events#create'
+  get '/event/new', to: 'events#new', as: 'new_event'
+  get '/event/:id', to: 'events#show', as: 'event'
+  get '/event/:id/edit', to: 'events#edit', as: 'edit_event'
+  put '/event/:id', to: 'events#update'
+  patch '/event/:id', to: 'events#update'
+  delete '/event/:id', to: 'events#destroy'
```
Comment (juperameza) [inline config/routes.rb]: Checkout the rails source in routes to leave this more readable 
https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default

File: test/controllers/events_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class EventsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (juperameza) [inline test/controllers/events_controller_test.rb]: If you dont need this delete it

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by juperameza — APPROVED** [review]

File: app/controllers/events_controller.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+# Class...
+class EventsController < ApplicationController
+  def index
+    @events = Event.all
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
```
Comment (MauricioNC) [inline app/controllers/events_controller.rb]: You are repeating this line many times. Instead, use a before_action filter to set the event.

You can see the documentation at:
https://guides.rubyonrails.org/action_controller_overview.html#filters

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 7]
```diff
@@ -6,5 +8,6 @@
   get "up" => "rails/health#show", as: :rails_health_check
```
Comment (MauricioNC) [inline config/routes.rb]: If this route doesn't be part of your project, please remove it.

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 7 above)
Comment (ErikGmz) [reply config/routes.rb]: Fixed!

File: app/controllers/events_controller.rb (hunk 6 above)
Comment (ErikGmz) [reply app/controllers/events_controller.rb]: Fixed!

**Review by MauricioNC — APPROVED** [review]
LGTM!

**Pull request merged**